test/tools/router_check/router.cc

```diff
@@ -0,0 +1,204 @@
+#include "test/tools/router_check/router.h"
+
+Json::ObjectPtr RouterCheckTool::loadJson(const std::string& config_json,
+                                          const std::string& schema) {
+
+  try {
+    // Json configuration
+    Json::ObjectPtr loader = Json::Factory::LoadFromFile(config_json);
+    loader->validateSchema(schema);
+    return loader;
+  } catch (const EnvoyException& ex) {
+    std::cerr << "config schema JSON load failed: " << config_json << std::endl;
+    std::cerr << ex.what() << std::endl;
+    return nullptr;
+  }
+}
+
+bool RouterCheckTool::initializeFromConfig(const std::string& router_config_json) {
+  // TODO(hennna): Allow users to load a full config and extract the route
+  // configuration from it.
+  Json::ObjectPtr loader = loadJson(router_config_json, Json::Schema::ROUTE_CONFIGURATION_SCHEMA);
+
+  if (loader != nullptr) {
+    config_.reset(new Router::ConfigImpl(*loader, runtime_, cm_, false));
+    return true;
+  }
+  return false;
+}
+
+bool RouterCheckTool::compareEntriesInJson(const std::string& expected_route_json) {
+  if (config_ == nullptr) {
+    return false;
+  }
+
+  // Load tool config json
+  Json::ObjectPtr loader = loadJson(expected_route_json, Json::ToolSchema::routerCheckSchema());
+
+  if (loader == nullptr) {
+    return false;
+  }
+
+  bool no_failures = true;
+  // Iterate through each test case
+  for (const Json::ObjectPtr& check_config : loader->asObjectArray()) {
+    // Load parameters from json
+    ToolConfig tool_config;
+    tool_config.parseFromJson(check_config);
```

Comment:
It seems folks around here aren't comfortable with exceptions being thrown in constructors. I've updated the code to follow the model in https://github.com/lyft/envoy/blob/master/source/common/network/cidr_range.h#L95 using static factory methods to create instances of ToolConfig and RouterCheckTool.